contracts/RelayHub.sol

```diff
@@ -287,7 +287,7 @@ contract RelayHub is IRelayHub {
 
         // We don't yet know how much gas will be used by the recipient, so we make sure there are enough funds to pay
         // for the maximum possible charge.
-        require(gasPrice * initialGas <= balances[recipient], "Recipient balance too low");
+        require(calculateCharge(initialGas, gasPrice, transactionFee) <= balances[recipient], "Recipient balance too low");
```

Comment:
Shouldn't it be `maxPossibleCharge(gasLimit, gasPrice, transactionFee) <= balances[recipient]`?

Reply:
I'm not sure, doesn't `initialGas` already account for the additional pre and post gas costs? L286 checks that `initialGas` is already over `requiredGas(gasLimit)`, where `requiredGas` adds all the overheads (`gasOverhead + gasReserve + acceptRelayedCallMaxGas + preRelayedCallMaxGas + postRelayedCallMaxGas`). Changing this into `maxPossibleCharge` would end up counting those overheads twice.

Either way, I've asked for another set of eyes to look into this, just in case.

Reply:
I've just noted I misread your suggestion: I had understood `maxPossibleCharge(initialGas, gasPrice, transactionFee)` instead of `maxPossibleCharge(gasLimit, gasPrice, transactionFee)`. It makes sense now, and I have asked around and there is agreement on it. Thanks Shahaf!